### DISCUSSION TITLE
Config error for unknown reasons.

### DISCUSSION THREAD

**Discussion by martadinata666** [stashapp/stash#4052] [General]
The stash itself served and work well, but today it suddenly fail to run due missing config.
```
stash-app-1  | INFO[2023-08-22 18:17:40] using config file: /home/docker/stash/config.yml 
stash-app-1  | panic: error initializing configuration: missing the following mandatory settings: generated [recovered]
stash-app-1  |  panic: Panic: error initializing configuration: missing the following mandatory settings: generated
```
My `config.yml`
```
create_image_clip_from_videos: false
dangerous_allow_public_without_auth: "false"
database: /home/docker/stash/stash-go.sqlite
gallery_cover_regex: (poster|cover|folder|board)\.[^\.]+$
generated: ""
host: 0.0.0.0
jwt_secret_key: 8442ff91ab41b83f07011b9e006495b337407822014aa7eb2f45fad289044172
no_proxy: localhost,127.0.0.1,192.168.0.0/16,10.0.0.0/8,172.16.0.0/12
nobrowser: true
notifications_enabled: true
parallel_tasks: 1
plugins_path: /home/docker/stash/plugins
port: 9999
preview_audio: true
preview_exclude_end: "0"
preview_exclude_start: "0"
preview_segment_duration: 0.75
preview_segments: 12
scrapers_path: /home/docker/stash/scrapers
security_tripwire_accessed_from_public_internet: ""
sequential_scanning: false
session_store_key: 44014441ce168bac3348ef211f1e37e7602e91703be61909318e6e93553dcdb4
show_one_time_moved_notification: true
sound_on_preview: false
theme_color: '#202b33'
write_image_thumbnails: true
```

Any clue why this happens, I can wipe the data dir and start anew, but I need to know what is causing this. 🤔

**Comment by MrX292** [comment]
You need to set the generated path. Probably /home/docker/stash/generated, but you would need to check it.

**Comment by martadinata666** [comment]
ah yes, that do the trick, but can't get any grasp on why it suddenly unset 🤔

**Comment by dingdongsolong4** [comment]
In a Docker setup, the generated path is usually overridden by the environment variable `STASH_GENERATED`: see https://github.com/stashapp/stash/blob/develop/docker/production/docker-compose.yml#L21.
If you're running in Docker, you need to make sure that your generated, metadata, blobs and cache paths are correctly mapped using volumes. Also check that sample `docker-compose.yml` file for reference.  If they aren't mapped properly, then if the docker container is restarted, you will lose your generated files, and if "Filesystem" blob storage is enabled, your scene covers and performer images as well.

If you're unsure, please ask for assistance on the #docker-help channel in the Discord server.

**Comment by martadinata666** [comment]
As stated it used to work well and run fine, but suddenly it throw an error about missing `generated` from some path to `""`. Or for some reason stash overwrite the `config.yml` from `STASH_GENERATED` that I don't set as the path already set in the configuration file.

**Comment by dingdongsolong4** [comment]
Well if it works, it works. Just trying to understand _why_ it happened, for in case it's a bug that needs fixing :)
"Sudden errors" usually shouldn't happen.

If you're not setting `STASH_GENERATED` in `docker-compose.yml`, then that explains it - without the override then stash will use the `config.yml` value, and `""` counts as missing in this case. But like I said, if you are running in Docker, you probably should be using the `STASH_GENERATED` override, because Docker requires the volumes to be mapped properly and if you set the path from within Stash then it's easy to forget to map the path, which will cause data loss when you restart the container.
